runtime-decompiler/src/main/java/org/jrd/frontend/frame/settings/SettingsView.java

```diff
@@ -55,43 +69,150 @@
             } else {
-                dir = new File(getJrdLocation() + File.separator + "decompiler_agent" + File.separator + "target");
+                dir = new File(Directories.getJrdLocation() + File.separator +
+                        "decompiler_agent" + File.separator + "target");
             }
-            chooser.setCurrentDirectory(fallback(dir));
+            chooser.setCurrentDirectory(FileSelectorArrayRow.fallback(dir));
 
             this.setLayout(new GridBagLayout());
             GridBagConstraints gbc = new GridBagConstraints();
             gbc.anchor = GridBagConstraints.WEST;
             gbc.fill = GridBagConstraints.BOTH;
 
-            gbc.insets = new Insets(0,20,0,0);
+            addAgent(gbc);
+
+            // Nested Jars
+            nestedJars = new JLabel("Nested Jars Settings:");
+            newExtensionsTextField = new JTextField();
+            newExtensionsTextField.addKeyListener(new KeyListener() {
+                @Override
+                public void keyTyped(KeyEvent keyEvent) {
+
+                }
+
+                @Override
+                public void keyPressed(KeyEvent keyEvent) {
+                    if (keyEvent.getKeyCode() == KeyEvent.VK_ENTER) {
+                        confirmExtensions();
+                    }
+                }
+
+                @Override
+                public void keyReleased(KeyEvent keyEvent) {
+
+                }
+            });
+
+            defaultListModel = new DefaultListModel<>();
+            currentExtensionsList = new JList<>(defaultListModel);
+            currentExtensionsList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+            currentExtensionsList.setLayoutOrientation(JList.VERTICAL);
+            currentExtensionsList.setVisibleRowCount(-1);
+
+            scrollPane = new JScrollPane(currentExtensionsList);
+            scrollPane.setPreferredSize(new Dimension(0, 200));
+
+            addButton = new JButton("Add");
+            addButton.addActionListener(actionEvent -> confirmExtensions());
+
+            removeButton = new JButton("Remove");
+            removeButton.addActionListener(actionEvent -> defaultListModel.removeElementAt(currentExtensionsList.getSelectedIndex()));
+
+            useDefaults = new JCheckBox("Use default extensions");
+            useDefaults.addItemListener(itemEvent -> {
+                newExtensionsTextField.setEnabled(!useDefaults.isSelected());
+                addButton.setEnabled(!useDefaults.isSelected());
+                removeButton.setEnabled(!useDefaults.isSelected());
+                currentExtensionsList.setEnabled(!useDefaults.isSelected());
+            });
+            useDefaults.setToolTipText(BytecodeDecompilerView.styleTooltip() +
+                    "Default extensions are: .zip, .jar, .war, .ear");
+
+            // Setup
+            List<String> defExtensions = ArchiveManagerOptions.getInstance().getExtensions();
```

Comment:
`getExtensions();` returns not only default extensions, but also currently set extensions. Therefore, something like 'currentExtensions' would be more proper.